doc/modules/preprocessing.rst

```diff
@@ -248,6 +256,66 @@ a :class:`KernelCenterer` can transform the kernel matrix
 so that it contains inner products in the feature space
 defined by :math:`phi` followed by removal of the mean in that space.
 
+.. _preprocessing_transformer:
+
+Non-linear transformation
+=========================
+
+Like scalers, :class:`QuantileTransformer` puts each feature into the same
+range or distribution. However, by performing a rank transformation, it smooths
+out unusual distributions and is less influenced by outliers than scaling
+methods. It does, however, distort correlations and distances within and across
+features.
+
+:class:`QuantileTransformer` and :func:`quantile_transform` provide a
+non-parametric transformation based on the quantile function to map the data to
+a uniform distribution with values between 0 and 1::
+
+  >>> from sklearn.datasets import load_iris
+  >>> from sklearn.model_selection import train_test_split
+  >>> iris = load_iris()
+  >>> X, y = iris.data, iris.target
+  >>> X_train, X_test, y_train, y_test = train_test_split(X, y)
+  >>> quantile_transformer = preprocessing.QuantileTransformer(
+  ...     smoothing_noise=1e-12)
+  >>> X_train_trans = quantile_transformer.fit_transform(X_train)
+  >>> X_test_trans = quantile_transformer.transform(X_test)
+
+  >>> np.percentile(X_train[:, 0], [0, 25, 50, 75, 100])
+  ... # doctest: +ELLIPSIS, +SKIP
+  array([...])
+  >>> np.percentile(X_train_trans[:, 0], [0, 25, 50, 75, 100])
+  ... # doctest: +ELLIPSIS, +SKIP
+  array([...])
+  >>> np.percentile(X_test[:, 0], [0, 25, 50, 75, 100])
+  ... # doctest: +ELLIPSIS, +SKIP
+  array([...])
+  >>> np.percentile(X_test_trans[:, 0], [0, 25, 50, 75, 100])
+  ... # doctest: +ELLIPSIS, +SKIP
+  array([...])
```

Comment:
I don't see the point of having docstests with "arrray([...])" as the output. I think we should display  the actual percentiles up to 2 digits. We need to fix the `random_state=0` value to ensure that the results are reproducible.